File: Same Tree/code.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool IsSameTree(TreeNode p, TreeNode q) {
        string pTraverse = Traverse(p);
            string qTraverse = Traverse(q);

            return pTraverse.Equals(qTraverse);
    }
    
    private string Traverse(TreeNode node)
        {
            if (node == null)
                return string.Empty;

            return string.Format("{0} {1} {2} ", Traverse(node.left), node.val, Traverse(node.right));
        }
}

File: Path Sum II/code.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ArrayList<ArrayList<Integer>> pathSum(TreeNode root, int sum) {
        ArrayList<ArrayList<Integer>> paths = new ArrayList<ArrayList<Integer>>();

            if (root != null)
                traceSum(root, 0, sum, paths, new ArrayList<Integer>());
            
            return paths;
    }
    
    public void traceSum(TreeNode node, int sum, int goalSum, ArrayList<ArrayList<Integer>> paths, ArrayList<Integer> currentPath)
        {
            sum += node.val;
            if ((node.left == null) && (node.right == null))
            {
                if (sum == goalSum)
                {
                    currentPath.add(node.val);
                    paths.add(currentPath);
                    return;
                }
                else
                {
                    return;
                }
            }
            else 
            {
                
                
                if (node.left != null)
                {
                    ArrayList<Integer> newPath = new ArrayList<Integer>();
                newPath.addAll(currentPath);
                newPath.add(node.val);
                
                    traceSum(node.left, sum, goalSum,paths, newPath);
                }
                if (node.right != null)
                {
                    ArrayList<Integer> newPath = new ArrayList<Integer>();
                newPath.addAll(currentPath);
                newPath.add(node.val);
                
                    traceSum(node.right, sum, goalSum, paths, newPath);
                }
            }
        }
}